segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/rolling/RollingStorage.java

```diff
@@ -442,6 +442,11 @@ public void concat(SegmentHandle targetHandle, long targetOffset, String sourceS
         LoggerHelpers.traceLeave(log, "concat", traceId, target, targetOffset, sourceSegment);
     }
 
+    @Override
+    public void rename(SegmentHandle handle, String newName) throws StreamSegmentException {
```

Comment:
Rolling storage is implemented using concept of chunks and a header file that points to them.

Proper implementation will involve following steps
```

Find name of the header file (see functions in nameutils)
Deserialize the header file
Find out list of chunks
 For each chunk in list { 
          parse chunk name to extract offset, 
          using offset+new segment name generate new chunk name
          rename by calling baseStorage.rename
          update name in the header data structure
}
serialize updated header (so that it points to the recently renamed chunks)
now rename the header itself to new name.
```

- Just calling this.baseStorage.rename won't work because there is no such segment in base segment.
- Just renaming the header won't work because then we are still pointing to the old chunks. When new segment is created with old name, it will try to create a chunk at offset 0 which will be already existing.